src/vm/functions/mod.rs

```diff
@@ -343,8 +343,16 @@ fn special_let(args: &[SymbolicExpression], env: &mut Environment, context: &Loc
             inner_context.variables.insert(binding_name, binding_value);
         }
 
-        // evaluate the let-body
-        eval(&args[1], env, &inner_context)
+        // evaluate the let-bodies
+        let mut results = Vec::new();
```

Comment:
Here, you'd want to escalate errors in the bodies before evaluating subsequent ones (and you can also avoid using a Vec by using an Option) e.g.:

```rust
let mut last_result = none;
for body in args[1..].iter() {
  body_result = eval(&body, env, &inner_context)?;
  last_result.replace(body_result);
}

last_result
  .ok_or_err(UncheckedError::InvalidArguments("Must pass at least 1 body to (let ...)".to_string()).into())
```

This way, if an arithmetic error or an `expects!` occurred in the first body the subsequent bodies _won't be processed_, which is probably what we want. We should add a test case for this behavior (and possibly a note in the documentation for `let`).

Reply:
🙏! 
Addessed with https://github.com/blockstack/blockstack-core/pull/1049/commits/b2bf90bb7cd0f18122d0e3f7d6aa7f07ca38ebe7